plugin/pkg/shoot/validator/admission_test.go

```diff
@@ -2439,6 +2494,30 @@ var _ = Describe("validator", func() {
 							Name:    imageName2,
 							Version: latestNonExpiredVersion,
 						}))
+						Expect(newShoot.Spec.Provider.Workers[2].Machine.Image).To(Equal(&core.ShootMachineImage{
+							Name:    imageName2,
+							Version: latestNonExpiredVersion,
+						}))
+					})
+
+					It("should default version of worker pool to latest non-preview version when machine architecture is changed", func() {
+						newShoot := shoot.DeepCopy()
+						newShoot.Spec.Provider.Workers[0].Machine.Type = "machine-type-3"
+						newShoot.Spec.Provider.Workers[0].Machine.Image = nil
+						newShoot.Spec.Provider.Workers[0].Machine.Architecture = pointer.String("arm64")
+
+						Expect(coreInformerFactory.Core().InternalVersion().Projects().Informer().GetStore().Add(&project)).To(Succeed())
+						Expect(coreInformerFactory.Core().InternalVersion().CloudProfiles().Informer().GetStore().Add(&cloudProfile)).To(Succeed())
+						Expect(coreInformerFactory.Core().InternalVersion().Seeds().Informer().GetStore().Add(&seed)).To(Succeed())
+						attrs := admission.NewAttributesRecord(newShoot, &shoot, core.Kind("Shoot").WithVersion("version"), newShoot.Namespace, newShoot.Name, core.Resource("shoots").WithVersion("version"), "", admission.Update, &metav1.CreateOptions{}, false, nil)
+
+						err := admissionHandler.Admit(context.TODO(), attrs, nil)
+
+						Expect(err).NotTo(HaveOccurred())
```

Comment:
```suggestion
						Expect(admissionHandler.Admit(context.TODO(), attrs, nil)).To(Succeed())
```

(I know it's not this way for the other cases, but for newly added cases we can improve readability by following this scheme).